### PR TITLE
Confluent support new version 2021-03-01-preview

### PR DESCRIPTION
**Pull request by alluri02**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [x] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by JonathanCrd** [comment]
Hello, could you please provide the link to the Azure Rest APi Specs' PR?

File: sdk/confluent/Microsoft.Azure.Management.Confluent/src/Generated/Models/ConfluentAgreementResource.cs [hunk 1]
```diff
@@ -48,11 +50,12 @@ public ConfluentAgreementResource()
         /// <param name="signature">Terms signature.</param>
         /// <param name="accepted">If any version of the terms have been
         /// accepted, otherwise false.</param>
-        public ConfluentAgreementResource(string id = default(string), string name = default(string), string type = default(string), string publisher = default(string), string product = default(string), string plan = default(string), string licenseTextLink = default(string), string privacyPolicyLink = default(string), System.DateTime? retrieveDatetime = default(System.DateTime?), string signature = default(string), bool? accepted = default(bool?))
+        public ConfluentAgreementResource(string id = default(string), string name = default(string), string type = default(string), SystemData systemData = default(SystemData), string publisher = default(string), string product = default(string), string plan = default(string), string licenseTextLink = default(string), string privacyPolicyLink = default(string), System.DateTime? retrieveDatetime = default(System.DateTime?), string signature = default(string), bool? accepted = default(bool?))
```
Comment (JonathanCrd) [inline sdk/confluent/Microsoft.Azure.Management.Confluent/src/Generated/Models/ConfluentAgreementResource.cs:53]: This is a breaking change; new parameters should be at the end. Please rearrange the parameters in the swagger.

File: sdk/confluent/Microsoft.Azure.Management.Confluent/src/Generated/Models/UserDetail.cs [hunk 2]
```diff
@@ -30,10 +30,10 @@ public UserDetail()
         /// <summary>
         /// Initializes a new instance of the UserDetail class.
         /// </summary>
+        /// <param name="emailAddress">Email address</param>
         /// <param name="firstName">First name</param>
         /// <param name="lastName">Last name</param>
-        /// <param name="emailAddress">Email address</param>
-        public UserDetail(string firstName = default(string), string lastName = default(string), string emailAddress = default(string))
+        public UserDetail(string emailAddress, string firstName = default(string), string lastName = default(string))
```
Comment (JonathanCrd) [inline sdk/confluent/Microsoft.Azure.Management.Confluent/src/Generated/Models/UserDetail.cs:36]: Here is another breaking change. `string emailAddress` was rearrange, please move it back to the end by modifying the order in the swagger file.

**Review by JonathanCrd — CHANGES_REQUESTED** [review]

File: sdk/confluent/Microsoft.Azure.Management.Confluent/src/Generated/Models/UserDetail.cs (hunk 2 above)
Comment (alluri02) [reply sdk/confluent/Microsoft.Azure.Management.Confluent/src/Generated/Models/UserDetail.cs:36]: Hi @JonathanCrd, the Email field is at the end only in swagger - 
https://github.com/Azure/azure-rest-api-specs/blob/f69f2e8053ac8ae352b11c33655234dea4cf4cfd/specification/confluent/resource-manager/Microsoft.Confluent/preview/2021-03-01-preview/confluent.json#L813

File: sdk/confluent/Microsoft.Azure.Management.Confluent/src/Generated/Models/ConfluentAgreementResource.cs (hunk 1 above)
Comment (alluri02) [reply sdk/confluent/Microsoft.Azure.Management.Confluent/src/Generated/Models/ConfluentAgreementResource.cs:53]: Hi @JonathanCrd, SystemData is at the end only - https://github.com/Azure/azure-rest-api-specs/blob/f69f2e8053ac8ae352b11c33655234dea4cf4cfd/specification/confluent/resource-manager/Microsoft.Confluent/preview/2021-03-01-preview/confluent.json#L585

**Comment by alluri02** [comment]
> Hello, could you please provide the link to the Azure Rest APi Specs' PR?
Sorry forgot to put the link - 
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/confluent/resource-manager/Microsoft.Confluent/preview/2021-03-01-preview/confluent.json

**Review by JonathanCrd — APPROVED** [review]
LGTM

**Pull request merged**